recon/src/protocol.rs

```diff
@@ -44,15 +48,58 @@ use crate::{
 // Even a small limit will quickly mean that both peers have work to do.
 const PENDING_RANGES_LIMIT: usize = 20;
 
+// The max number of writes we'll batch up before flushing anything to disk.
+// As we descend the tree and find smaller ranges, this won't apply as we have to flush
+// before recomputing a range, but it will be used when we're processing large ranges we don't yet have.
+const INSERT_BATCH_SIZE: usize = 100;
```

Comment:
so I notice this moved from `recon/src/libp2p/protocol.rs` to `recon/src/protocol.rs`.  How should we be thinking about these two `protocol.rs` files and what belongs in one vs the other?

Reply:
recon is "lower".. so I pushed it down to avoid having to pipe it through both levels. the recon protocol is the set reconciliation part, while the libp2p one includes the network streams and is an entry point from the libp2p connection. I think it would be nice to review things and see what we can simplify and if we still want/need all the layers.